hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/ReadBufferManager.java

```diff
@@ -502,11 +503,66 @@ int getCompletedReadListSize() {
     return completedReadList.size();
   }
 
+  @VisibleForTesting
+  public LinkedList<ReadBuffer> getCompletedReadListCopy() {
+    return new LinkedList<>(completedReadList);
+  }
+
+  @VisibleForTesting
+  public LinkedList<Integer> getFreeListCopy() {
```

Comment:
nit, just make simple list. Now, small risk of that copy happening while something else changes. Now it is only test code, but could lead to intermittent test fails.

so: make these all synchronized; return List() and use ArrayList inside